go/vt/vtadmin/cluster/cluster.go

```diff
@@ -1030,17 +1103,21 @@ func (c *Cluster) GetSrvVSchemas(ctx context.Context, cells []string) ([]*vtadmi
 		return nil, fmt.Errorf("Vtctld.Dial(cluster=%s) failed: %w", c.ID, err)
 	}
 
-	res, err := c.Vtctld.GetSrvVSchemas(ctx, &vtctldatapb.GetSrvVSchemasRequest{
+	if err := c.topoReadPool.Acquire(ctx); err != nil {
+		return nil, fmt.Errorf("GetSrvVSchema(cluster = %s, cells = %v) failed to acquire topoReadPool: %w", c.ID, cells, err)
+	}
+	resp, err := c.Vtctld.GetSrvVSchemas(ctx, &vtctldatapb.GetSrvVSchemasRequest{
 		Cells: cells,
 	})
+	c.topoReadPool.Release()
```

Comment:
Non-blocking question -- in some cases, we use `defer c.topoReadPool.Relase()` and in others, like here, we Acquire and the Release sequentially (for lack of a better term). 

In this case, it would be equivalent to `defer`, since it's the only Acquire/Release in the function, right? Is there much of a difference? (You've kinda sold me on the `defer` way since it seems easy (for me) to forget, otherwise, so I'm curious. 😈)

Reply:
This is a general comment not directed at this specific usage. Until Go 1.14, defer had a significant penalty for hot paths, so you saw a lot of people avoiding it. That has mostly been eliminated, so except for rare circumstances, defer is the safer choice. IMO, it's easier to read, plus eliminates the possibility of adding a return somewhere without handling the unlock.
https://rakyll.org/inlined-defers/

Reply:
Yeah, that's exactly the intent I had behind this. If we're just making the rpc, and moving the result from one struct to another (so no other function calls, or iterating over elements in a response, etc), then it's "fine" to just defer the release, but if we're going to do any other "meaningful" work (which i am loosely defining as function calls or iterations) then we should release immediately after finishing the rpc to free up other waiting goroutines.

Reply:
Ah somehow ... github did not show me Derek's comment. Derek's completely correct, but there's another consideration in that we are not _just_ making the rpc call, so it doesn't make sense to be holding onto a slot in the pool for us to do unrelated work.